server/src/main/java/org/opensearch/index/store/lockmanager/RemoteBufferedIndexOutput.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.OutputStreamIndexOutput;
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.index.store.RemoteIndexOutput;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Blob;
+
+/**
+ * Class for output to a file in a {@link RemoteBufferedOutputDirectory}. Used for all output operations to the remote store.
+ * @see RemoteBufferedOutputDirectory
+ *
+ * @opensearch.internal
+ */
+public class RemoteBufferedIndexOutput extends RemoteIndexOutput {
```

Comment:
This can be taken out of `lockmanager` package as the use case could be generic.

Reply:
ack